packages/ai-jsx/src/core/conversation.tsx

```diff
@@ -309,3 +323,204 @@ export async function* ShowConversation(
   // we can indicate that we've already yielded the final frame.
   return AI.AppendOnlyStream;
 }
+
+/**
+ * @hidden
+ * "Shrinks" a conversation messages according to a cost function (i.e. token length),
```

Comment:
Ha, I originally wrote "e.g." and convinced myself that it was too performative because there's no other reasonable cost function. Will change it back to "e.g." :)